File: KahveQRKodOlusturucuJava-calismalarim/app/src/main/java/com/example/qrkodolusturucujava/MainActivity.java
package com.example.qrkodolusturucujava;

import android.graphics.Color;
import android.os.Bundle;
import android.text.InputType;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

import com.example.qrkodolusturucujava.databinding.ActivityMainBinding;
import com.example.qrkodolusturucujava.viewmodel.MainVM;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private MainVM viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        viewModel = new ViewModelProvider(this).get(MainVM.class);

        observeViewModel();

        binding.btnQRKod.setOnClickListener(v -> {
            binding.btnQRKod.setBackgroundColor(getResources().getColor(R.color.btnSeciliBackground, getTheme()));
            binding.btnUrunlerimiz.setBackgroundColor(Color.TRANSPARENT);
            toQRKodFragment();
        });

        binding.btnUrunlerimiz.setOnClickListener(v -> {
            binding.btnUrunlerimiz.setBackgroundColor(getResources().getColor(R.color.btnSeciliBackground, getTheme()));
            binding.btnQRKod.setBackgroundColor(Color.TRANSPARENT);
            toUrunlerFragment();
        });

        if (!viewModel.checkTelefonNumarasi()) {
            showInputDialog();
        }

        viewModel.getTelefonNumarasiLiveData().observe(this, numara -> {
            Log.e("MainVMActivity", "Gelen Telefon Numarası: " + numara);
            if (numara == null) {
                showInputDialog();
            } else {
                viewModel.kullaniciEkle(numara);
            }
        });
    }

    private void observeViewModel() {
        viewModel.getHediyeKahveLiveData().observe(this, hediyeKahve ->
                binding.tvHediyeKahve.setText("Hediye Kahve: " + hediyeKahve));

        viewModel.getKahveSayisiLiveData().observe(this, kahveSayisi ->
                binding.tvKullanimlar.setText("Kullanımlar: " + kahveSayisi + "/5"));
    }

    private void showInputDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("Bilgi Girişi");

        EditText input = new EditText(this);
        input.setHint("Lütfen telefon numaranızı giriniz");
        input.setInputType(InputType.TYPE_CLASS_NUMBER);
        builder.setView(input);

        builder.setPositiveButton("Tamam", (dialog, which) -> {
            String userInput = input.getText().toString().trim();
            if (!userInput.isEmpty()) {
                viewModel.numarayiKaydet(userInput);
                Toast.makeText(this, "Numara kaydedildi!", Toast.LENGTH_SHORT).show();
                toQRKodFragment();
            }
            dialog.dismiss();
        });

        builder.setNegativeButton("İptal", (dialog, which) -> {
            finish();
            dialog.cancel();
        });

        builder.show();
    }

    private void toQRKodFragment() {
        View navHost = findViewById(R.id.fragmentContainerView);
        NavController navController = Navigation.findNavController(navHost);
        navController.navigate(R.id.QRKodOlusturFragment);
    }

    private void toUrunlerFragment() {
        View navHost = findViewById(R.id.fragmentContainerView);
        NavController navController = Navigation.findNavController(navHost);
        navController.navigate(R.id.urunlerimizFragment2);
    }
}

File: KahveQrOlusturucuKotlin-calismalarim/app/src/main/java/com/example/qrkodolusturucu/MainActivity.kt
package com.example.qrkodolusturucu

import android.content.pm.PackageManager
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.telephony.TelephonyManager
import android.text.InputType
import android.util.Log
import android.widget.*
import androidx.activity.viewModels
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import com.example.qrkodolusturucu.data.numara_kayit.NumaraKayitSp
import com.example.qrkodolusturucu.databinding.ActivityMainBinding
import com.example.qrkodolusturucu.services.retrofitservice.KahveHTTPServisDao
import com.example.qrkodolusturucu.viewmodel.MainVM
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private val viewModel: MainVM by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        observeViewModel()

        binding.btnQRKod.setOnClickListener {
            binding.btnQRKod.setBackgroundColor(resources.getColor(R.color.btnSeciliBackground, theme))
            binding.btnUrunlerimiz.setBackgroundColor(Color.TRANSPARENT)
            toQRKodFragment()
        }

        binding.btnUrunlerimiz.setOnClickListener {
            binding.btnUrunlerimiz.setBackgroundColor(resources.getColor(R.color.btnSeciliBackground, theme))
            binding.btnQRKod.setBackgroundColor(Color.TRANSPARENT)
            toUrunlerFragment()
        }

        observeViewModel()
        if (!viewModel.checkTelefonNumarasi()){
            showInputDialog()
        }


        viewModel.telefonNumarasiLiveData.observe(this) { numara ->
            Log.e("MainVMActivity", "Gelen Telefon Numarası: $numara")  // Burada numarayı logluyoruz
            if (numara == null) {
                showInputDialog()
            } else {
                viewModel.kullaniciEkle(numara)
            }
        }
    }

    private fun observeViewModel() {
        viewModel.hediyeKahveLiveData.observe(this) { hediyeKahve ->
            binding.tvHediyeKahve.text = "Hediye Kahve: $hediyeKahve"
        }

        viewModel.kahveSayisiLiveData.observe(this) { kahveSayisi ->
            binding.tvKullanimlar.text = "Kullanımlar: $kahveSayisi/5"
        }
    }

    private fun showInputDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Bilgi Girişi")

        val input = EditText(this)
        input.setHint("Lütfen telefon numaranızı giriniz")
        input.inputType = InputType.TYPE_CLASS_NUMBER
        builder.setView(input)

        builder.setPositiveButton("Tamam") { dialog, _ ->
            val userInput = input.text.toString().trim()
            if (userInput.isNotEmpty()) {
                viewModel.numarayiKaydet(userInput)
                Toast.makeText(this, "Numara kaydedildi!", Toast.LENGTH_SHORT).show()
                toQRKodFragment()
            }
            dialog.dismiss()
        }

        builder.setNegativeButton("İptal") { dialog, _ ->
            finish()
            dialog.cancel()
        }

        builder.show()
    }

    private fun toQRKodFragment() {
        val navController = supportFragmentManager.findFragmentById(R.id.fragmentContainerView)?.findNavController()
        navController?.navigate(R.id.QRKodOlusturFragment)
    }

    private fun toUrunlerFragment() {
        val navController = supportFragmentManager.findFragmentById(R.id.fragmentContainerView)?.findNavController()
        navController?.navigate(R.id.urunlerimizFragment2)
    }
}
